modules/web/src/com/haulmont/cuba/web/app/ui/core/settings/SettingsWindow.java

```diff
@@ -183,8 +183,12 @@ public void init(Map<String, Object> params) {
                 .withCaption(getMessage("resetScreensSettings"))
                 .withDescription(getMessage("resetScreensSettings.description"))
                 .withHandler(actionPerformedEvent -> {
-                    userSettingService.deleteAllScreensSettings(ClientType.WEB);
-                    showNotification(getMessage("resetScreensSettings.notification"));
+                    if (getWindowManager().getOpenWindows().size() > 1) {
```

Comment:
I think we need confirmation dialog here with description what kind of settings will be removed